packages/eslint-plugin/src/rules/return-await.ts

```diff
@@ -169,11 +169,24 @@ export default util.createRule({
         return;
       }
 
+      // any could be thenable; do not auto-fix
+      const useAutoFix = !util.isTypeAnyType(type);
+      const removeFix = (fixer: TSESLint.RuleFixer): TSESLint.RuleFix | null =>
+        removeAwait(fixer, node);
       if (isAwait && !isThenable) {
         context.report({
```

Comment:
I'd slightly prefer these variables to be inside the below if statement but that is pretty minor.  

Probably want to use suggestion for `unknown` also since [`unknown | Promise<unknown>` collapses to just `unknown`](https://www.typescriptlang.org/play?#code/FAQwzgngdgxgBAMwK6wC4EsD2VEAoAeAXHCgNZSYDuOAPnAAoBOmAtumAKYA8ZF1AfAEoA3sDji4jDqiSMcISiHSo4+ANzAAvkA) so auto removing await there might also be undesirable. 

```suggestion
      if (isAwait && !isThenable) {
        // `any` or `unknown` could be thenable; do not auto-fix
        const useAutoFix = !(util.isTypeAnyType(type) || util.isTypeUnknownType(type));
        const removeFix = (fixer: TSESLint.RuleFixer): TSESLint.RuleFix | null =>
          removeAwait(fixer, node);
        context.report({
```

Reply:
Good point; change made (test added).